test/test_core.py

```diff
@@ -9098,6 +9098,12 @@ def test_asan(self, name, expected_output, cflags=None):
                  expected_output=expected_output, assert_all=True,
                  check_for_error=False, assert_returncode=NON_ZERO)
 
+  def test_asan_mimalloc(self):
+    # Run one of the asan tests in mimalloc (leaving the main coverage for
+    # the default allocator).
+    self.set_setting('MALLOC', 'mimalloc')
+    self.test_asan_use_after_free_c()
```

Comment:
I think it a good idea to call one `test_xx` method from another.   It just feels wrong somehow.

Maybe you can parameterize the `test_asan_use_after_free_c` test or create a separate helper they both can then call.

Also if this is a specific ASAN test maybe it should run in other rather than core?

Reply:
Either way sgtm, so I did it as you suggested, in the parameterization of the previous test. It does duplicate a tiny bit of that test now rather than reusing it though which is why I initially did it otherwise.

I think it makes sense in core for the same reason all the asan tests do, that they can depend on optimizations in various ways (due to interesting linking and JS stuff etc.). Maybe we don't need 100% of them in core, however, but at least some amount makes sense there.